config.go

```diff
@@ -333,3 +321,112 @@ func (config Configuration) String() string {
 	}
 	return buf.String()
 }
+
+func toUsage(usages []string) alpm.Usage {
+	if len(usages) == 0 {
+		return alpm.UsageAll
+	}
+
+	var ret alpm.Usage = 0
+	for _, usage := range usages {
+		switch usage {
+		case "Sync":
+			ret |= alpm.UsageSync
+		case "Search":
+			ret |= alpm.UsageSearch
+		case "Install":
+			ret |= alpm.UsageInstall
+		case "Upgrade":
+			ret |= alpm.UsageUpgrade
+		case "All":
+			ret |= alpm.UsageAll
+		}
+	}
+
+	return ret
+}
+
+func configureAlpm(conf *pacmanconf.Config) error {
```

Comment:
Passing the configuration to go-alpm and removing go-alpm's internal parser is not an option?

I would tag the current version of go-alpm as the last with an internal parser, import `go-pacmanconf` in `go-alpm` and avoid this alpm config in `yay, give me your thoughts on it

Reply:
Read your mid sentence comment here `It is now safe to drop all the config parsing from go-alpm.`, I guess we agree on this, just the boilerplate code is my question, wouldn't it still be useful for users of `go-alpm`

Reply:
Well The way I see it go-alpm should strive to just be go bindings for alpm. The config belongs to pacman and has nothing to do with alpm. For our use as a pacman wrapper it's useful to have config parsing.

With go-pacmanconf as a separate library you can still get config parsing without needed to write a ton of code.

Plus doing it this way, all out dependencies have no dependencies, it's just yay that depends on the individual pieces.

Reply:
Fair enough, I'll go and tag go-alpm still, so the parser removal is well documented

Reply:
I think you may have forgotten this isn't merged yet.

Reply:
Anyway I think this solution might actually be less code. It's +1000 but around 670 of those lines are the licence. Not sure now much the parsing code in alpm is though.